presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -13,26 +13,73 @@
  */
 package com.teradata.prestomanager.agent;
 
+import com.google.common.io.ByteStreams;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
 import java.util.concurrent.TimeUnit;
 
-import static java.lang.Runtime.getRuntime;
 import static java.lang.String.format;
 
-// TODO: Add logger
 public final class CommandExecutor
 {
-    private CommandExecutor() {}
+    private static final Logger LOGGER = LogManager.getLogger(CommandExecutor.class);
+    private static final int defaultTimeout = 60;
+    private final String[] commandArray;
+    private final int timeout;
+
+    private CommandExecutor(String[] commandArray, int timeout)
+            throws PrestoManagerException
+    {
+        if (commandArray.length == 0) {
+            throw new PrestoManagerException("Command array is empty");
```

Comment:
This should most likely be an `IllegalArgumentException`.
With both of my two new comments fixed, I approve this PR.